optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/change/DefaultProblemChangeDirector.java

```diff
@@ -0,0 +1,107 @@
+        Objects.requireNonNull(variableName, "Planning variable name (" + variableName + ") cannot be null.");
+        Objects.requireNonNull(entityConsumer, "Entity consumer (" + entityConsumer + ") cannot be null.");
+        Entity workingEntity = lookUpWorkingObject(entity);
+        scoreDirector.beforeVariableChanged(workingEntity, variableName);
+        entityConsumer.accept(workingEntity);
+        scoreDirector.afterVariableChanged(workingEntity, variableName);
+    }
+
+    @Override
+    public <ProblemFact> void addProblemFact(ProblemFact problemFact, Consumer<ProblemFact> problemFactConsumer) {
+        Objects.requireNonNull(problemFact, "Problem fact (" + problemFact + ") cannot be null.");
+        Objects.requireNonNull(problemFactConsumer, "Problem fact consumer (" + problemFactConsumer
+                + ") cannot be null.");
+        scoreDirector.beforeProblemFactAdded(problemFact);
+        problemFactConsumer.accept(problemFact);
+        scoreDirector.afterProblemFactAdded(problemFact);
+    }
+
+    @Override
+    public <ProblemFact> void removeProblemFact(ProblemFact problemFact, Consumer<ProblemFact> problemFactConsumer) {
+        Objects.requireNonNull(problemFact, "Problem fact (" + problemFact + ") cannot be null.");
+        Objects.requireNonNull(problemFactConsumer, "Problem fact consumer (" + problemFactConsumer
+                + ") cannot be null.");
+        ProblemFact workingProblemFact = lookUpWorkingObject(problemFact);
+        scoreDirector.beforeProblemFactRemoved(workingProblemFact);
+        problemFactConsumer.accept(workingProblemFact);
+        scoreDirector.afterProblemFactRemoved(workingProblemFact);
+    }
+
+    @Override
+    public <EntityOrProblemFact> void changeProblemProperty(EntityOrProblemFact problemFactOrEntity,
+            Consumer<EntityOrProblemFact> problemFactOrEntityConsumer) {
+        Objects.requireNonNull(problemFactOrEntity, "Problem fact or entity (" + problemFactOrEntity + ") cannot be null.");
+        Objects.requireNonNull(problemFactOrEntityConsumer, "Problem fact or entity consumer ("
+                + problemFactOrEntityConsumer + ") cannot be null.");
+        EntityOrProblemFact workingEntityOrProblemFact = lookUpWorkingObject(problemFactOrEntity);
+        scoreDirector.beforeProblemPropertyChanged(workingEntityOrProblemFact);
+        problemFactOrEntityConsumer.accept(workingEntityOrProblemFact);
+        scoreDirector.afterProblemPropertyChanged(workingEntityOrProblemFact);
+    }
+
+    @Override
+    public <EntityOrProblemFact> EntityOrProblemFact lookUpWorkingObject(EntityOrProblemFact externalObject) {
+        return scoreDirector.lookUpWorkingObject(externalObject);
+    }
+
+    @Override
+    public <EntityOrProblemFact> Optional<EntityOrProblemFact>
+            lookUpWorkingObjectOptionally(EntityOrProblemFact externalObject) {
+        return Optional.of(scoreDirector.lookUpWorkingObjectOrReturnNull(externalObject));
```

Comment:
```suggestion
        return Optional.ofNullable(scoreDirector.lookUpWorkingObjectOrReturnNull(externalObject));
```

Otherwise this is a guaranteed NPE.